### DISCUSSION TITLE
How to set Pump characteristics

### DISCUSSION THREAD

**Discussion by ZackeyZAKI** [OpenModelica/OpenModelica#10720] [Q&A] [answered]
When executing Modelica.Fluid.Machines.Pump with coefficient inputs on OMEdit, the following errors are generated:

[Modelica.Fluid.Machines: 328:5-329:76]: Function parameter V_flow_nominal was not given by the function call, and does not have a default value.
[Modelica.Fluid.Machines: 328:5-329:76]: Function parameter head_nominal was not given by the function call, and does not have a default value.

How should I input the characteristics?

**Comment by casella** [comment]
You can't execute the Pump model by itself, you should build a model that provides some boundary conditions, and then set all the parameters.

**Reply by ZackeyZAKI** [reply]
Thank you.

If the fluid is water and only the mass flow rate needs to be defined, would it be sufficient to define the characteristics of such a pump?

**Comment by casella** [comment, marked as answer]
The Pump model includes a physical model of the relationship between rotational speed, head, and flow rate. If you want to use that, you need to define the nominal operating curve by giving some point of it.

Otherwise you can use the ControlledPump, which takes a mass flow rate set point and just enforces it. This roughly corresponds to a speed-controlled volumetric pump, or to an pump with a very good mass flow rate feedback control system
